presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -801,6 +842,42 @@ private BackgroundHiveSplitLoader backgroundHiveSplitLoader(
                 Optional.empty());
     }
 
+    private BackgroundHiveSplitLoader backgroundHiveSplitLoader(
+            String location,
+            ConnectorSession connectorSession)
```

Comment:
If we follow the existing pattern, then we would add `String location` parameter to the existing `private BackgroundHiveSplitLoader backgroundHiveSplitLoader(HdfsEnvironment hdfsEnvironment, TupleDomain<HiveColumnHandle> compactEffectivePredicate, DynamicFilter dynamicFilter, Duration dynamicFilteringProbeBlockingTimeout, Optional<HiveBucketFilter> hiveBucketFilter, Table table, Optional<HiveBucketHandle> bucketHandle, Optional<ValidWriteIdList> validWriteIds)` and delegate to it in this function.
Then add yet another version of `backgroundHiveSplitLoader` with previous arguments to avoid changing other usages. This is definitely getting too verbose, maybe creating a BackgroundHiveSplitLoader builder would help to simplify this ?
We could avoid the table name discrepancy by using `new HivePartition(new SchemaTableName("testSchema", table.getTableName()))` instead of hard-coding `table_name`

Reply:
yup, adding a builder seems reasonable at this point, it'll also make easier to add future tests.

Reply:
Thanks for these suggestions, have gone with builder pattern as suggested.